sim_pipeline/gg_lens_pop.py

```diff
@@ -0,0 +1,182 @@
+        return num_sources_tested_mean
+
+    def get_num_sources_tested(self, testarea=None, num_sources_tested_mean=None):
+        """
+        Draw a realization of the expected distribution (Poisson) around the mean
+        for the number of source galaxies tested.
+        """
+        if num_sources_tested_mean is None:
+            num_sources_tested_mean = self.get_num_sources_tested_mean(testarea)
+        num_sources_range = np.random.poisson(lam=num_sources_tested_mean)
+        return num_sources_range
+
+    def draw_population(self, kwargs_lens_cuts):
+        """
+        return full population list of all lenses within the area
+        # TODO: need to implement a version of it. (improve the algorithm)
+            Draw a population of galaxy-galaxy lenses within the area.
+
+        :param kwargs_lens_cuts: validity test keywords
+        :type kwargs_lens_cuts: dict
+        :return: List of GGLens instances with parameters of the deflectors and lens and source light.
+        :rtype: list
+        """
+
+        # Initialize an empty list to store the GGLens instances
+        gg_lens_population = []
+        # Estimate the number of lensing systems
+        num_lenses = self._lens_galaxies.deflector_number()
+        # num_sources = self._source_galaxies.galaxies_number()
+        #        print(num_sources_tested_mean)
+        #        print("num_lenses is " + str(num_lenses))
+        #        print("num_sources is " + str(num_sources))
+        #        print(np.int(num_lenses * num_sources_tested_mean))
+
+        # Draw a population of galaxy-galaxy lenses within the area.
+        for _ in range(num_lenses):
+            lens = self._lens_galaxies.draw_deflector()
+            test_area = draw_test_area(deflector=lens)
+            num_sources_range = self.get_num_sources_tested(testarea=test_area)
+            # TODO: to implement this for a multi-source plane lens system
+            if num_sources_range > 0:
+                n = 0
+                while n < num_sources_range:
+                    source = self._sources.draw_source()
+                    gg_lens = GGLens(deflector_dict=lens, source_dict=source, cosmo=self.cosmo, test_area=test_area,
+                                     source_type=self._source_model_type)
+                    # Check the validity of the lens system
+                    if gg_lens.validity_test(**kwargs_lens_cuts):
+                        gg_lens_population.append(gg_lens)
+                        n = num_sources_range
+                    else:
+                        n += 1
+        return gg_lens_population
+
+
+def draw_test_area(deflector):
+    """
+    draw a test area around the deflector
+
+    :param deflector: deflector dictionary
+    :return: test area in arcsec^2
+    """
+    theta_e_infinity = theta_e_when_source_infinity(deflector)
+    test_area = np.pi * (theta_e_infinity * 1.3) ** 2
```

Comment:
Why the 1.3 value?

Reply:
The factor should more be like 2. So it is effectively the area within we can expect multiple images to occur, and be conservative with the area such that all possible multiply imaged configurations land in this area.

Reply:
I would change it to 2.5 and then comment on it as you did here @nkhadka21. Thanks a lot!

Reply:
Thanks Simon. I changed it to 2.5. I will make a PR with all the changes later.